website/server/models/newsPost.js

```diff
@@ -0,0 +1,73 @@
+import mongoose from 'mongoose';
+import baseModel from '../libs/baseModel';
+
+const { Schema } = mongoose;
+
+export const schema = new Schema({
+  title: { $type: String },
+  author: { $type: String },
+  credits: { $type: String },
+  publishDate: { $type: Date },
+  published: { $type: Boolean },
+  text: { $type: String },
+}, {
+  strict: true,
+  minimize: false, // So empty objects are returned
+  typeKey: '$type', // So that we can use fields named `type`
+});
+
+schema.plugin(baseModel, {
+  noSet: ['_id'],
+  timestamps: true,
+});
+
+schema.statics.getNews = async function getNews (isAdmin) {
+  let posts = [];
+  if (!isAdmin) {
+    posts = this.find({
+      published: true,
+      publishDate: { $lte: Date() },
+    })
+      .select('title publishDate credits text');
+  } else {
+    posts = this.find();
+  }
+  return posts.sort({ publishDate: -1 });
```

Comment:
I think `exec` is missing, without it you don't get a full promise
```suggestion
  return posts.sort({ publishDate: -1 }).exec();
```